src/controllers/project.js

```diff
@@ -1,132 +1,131 @@
+      const error = new Error('Could not find any Project.');
+      error.statusCode = 404;
+      throw error;
+    }
+    res.status(200).json({ projects });
+  } catch (err) {
+    if (!err.statusCode) {
+      err.statusCode = 500;
     }
+    next(err);
+  }
 };
 
 const getProjectByID = async (req, res, next) => {
-    const {id} = req.params;
-    const project = await Project.findById(id);
-    try {
-        if (!project) {
-        const error = new Error("Could not find Project.");
-        error.statusCode = 404;
-        throw error;
-        }
-        res.status(200).json({ project });
-    } catch (err) {
-        if (!err.statusCode) {
-        err.statusCode = 500;
-        }
-        next(err);
+  const { id } = req.params;
+  const project = await Project.findById(id);
+  try {
+    if (!project) {
+      const error = new Error('Could not find Project.');
+      error.statusCode = 404;
+      throw error;
+    }
+    res.status(200).json({ project });
+  } catch (err) {
+    if (!err.statusCode) {
+      err.statusCode = 500;
     }
+    next(err);
+  }
 };
 
 const getAllComments = async (req, res, next) => {
-    const {id} = req.params;
-    const project = await Project.findById(id);
-    try {
-        if (!project) {
-        const error = new Error("Could not find Project.");
-        error.statusCode = 404;
-        throw error;
-        }
-        res.status(200).json({ comments: project.comments });
-    } catch (err) {
-        if (!err.statusCode) {
-        err.statusCode = 500;
-        }
-        next(err);
+  const { id } = req.params;
+  const project = await Project.findById(id);
+  try {
+    if (!project) {
+      const error = new Error('Could not find Project.');
+      error.statusCode = 404;
+      throw error;
     }
+    res.status(200).json({ comments: project.comments });
+  } catch (err) {
+    if (!err.statusCode) {
+      err.statusCode = 500;
+    }
+    next(err);
+  }
 };
 
 const getCommentByID = async (req, res, next) => {
-    const {id, commentid} = req.params;
-    const project = await Project.findById(id);
-    try {
-        if (!project) {
-        const error = new Error("Could not find Project.");
-        error.statusCode = 404;
-        throw error;
-        }
-        // 
-        const comment = await project.comments.findById(commentid);
-        if (!comment) {
-            const error = new Error("Could not find Comment.");
-            error.statusCode = 404;
-            throw error;
-            }
-        res.status(200).json({ comment });
-    } catch (err) {
-        if (!err.statusCode) {
-        err.statusCode = 500;
-        }
-        next(err);
+  const { id, commentid } = req.params;
+  const project = await Project.findById(id);
+  try {
+    if (!project) {
+      const error = new Error('Could not find Project.');
+      error.statusCode = 404;
+      throw error;
+    }
+    //
+    const comment = await project.comments.find({ 'comment._id': commentid });
```

Comment:
this should project.find() not project.comments.find() since comments is not a model but an embedded document